File: rb/features.rb
# IbanValidation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IbanValidationFeatures
  def self.make_feature(name)
    case name
    when "base"
      IbanValidationBaseFeature.new
    when "test"
      IbanValidationTestFeature.new
    else
      IbanValidationBaseFeature.new
    end
  end
end

File: py/ibanvalidation_sdk/features.py
# IbanValidation SDK feature factory

from ibanvalidation_sdk.feature.base_feature import IbanValidationBaseFeature
from ibanvalidation_sdk.feature.test_feature import IbanValidationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IbanValidationBaseFeature(),
        "test": lambda: IbanValidationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
